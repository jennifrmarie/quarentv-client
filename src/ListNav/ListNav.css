.watchlist_main {
    list-style: none;
    align-self: center;
    border: 2px solid white;
    margin: 5px;
    width: 100%;
    /* text-align: left; */
}

body {
    --scrollbarBG: rgb(11, 3, 39);
    --thumbBG: rgb(230, 224, 231);
  }
  body::-webkit-scrollbar {
    width: 15px;
  }
  body {
    scrollbar-width: thin;
    scrollbar-color: var(--thumbBG) var(--scrollbarBG);
  }
  body::-webkit-scrollbar-track {
    background: var(--scrollbarBG);
  }
  body::-webkit-scrollbar-thumb {
    background-color: var(--thumbBG) ;
    border-radius: 6px;
    border: 3px solid var(--scrollbarBG);
  }

  .DashboardList_list::-webkit-scrollbar-track {
      background: rgb(11, 3, 39);
  }
  .DashboardList_list::-webkit-scrollbar-thumb {
      background-color: white;
  }
.DashboardList_list {
    font-family: 'Gugi';
    letter-spacing: .5em;
    width: 80%;
    background: linear-gradient(
        to top,
        rgba(62, 8, 71, 0.979),
        rgba(216, 13, 131, 0.6)
      );
    overflow-x: hidden;
    width: 50%;
    margin: 0;
}

.DashboardListNav {
    height: 100vh;
    overflow: hidden;
    display: flex;
    width: 100%;
    background-color: rgb(255, 0, 76);
    color: white;
}

.scoreboard__list {
    list-style: none;
    width: 100%;
}

.scoreboard__title {
    margin-top: 100px;
    margin-left: 0px;
    
}
.scoreboard_section > .scoreboard__title {

    width: 92%;
    padding: 20px 10px;
    background: inherit;
    background-attachment: fixed;
  }
  .scoreboard_section > .scoreboard__title::before {
    content: "";
    width: 100%;
    height: 100%;
    background: inherit;
    background-attachment: fixed;
    -webkit-filter: blur(12px);
    filter: blur(12px);
  }

.scoreboard_section {
    font-family: Audiowide;
    display: block;
    letter-spacing: .5em;
    /* margin-left: 20px; */
    text-align: center;
    padding-right: 220px;
    justify-self: center;
    /* margin-bottom: 20px; */
    width: 50%;
    overflow: clipped;
    background-size: stretch;
    background: linear-gradient(
        to top,
        rgba(62, 8, 71, 0.979),
        rgba(216, 13, 131, 0.6)
      ), url('https://i.imgur.com/gHdH0or.png');
    background-repeat: no-repeat;
}

.watched_list {
    display: flex;
    width: 33%;
    margin: 0
}

h2 {

    font-size: 1.5em;
    margin-block-start: 0em;
    margin-block-end: 0em;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
}

.delete_button {
    float: right;
    margin-right: 20px;
    padding-top: 15px;
    
}
li.scoreboard__list {
    margin: 21px;
}

.list__title {
    font-size: calc(15px + .5vw);
    margin-bottom: 10px;
    margin-top: 10px;
}


@media only screen 
  and (min-device-width: 375px) 
  and (max-device-width: 812px) 
  and (-webkit-min-device-pixel-ratio: 3) { 

.DashboardListNav {
    display: flex;
    flex-direction: column;
    
}

.scoreboard_section > .scoreboard__title {

    width: 100%;
    /* padding: 20px 10px; */
    background: inherit;
    position: absolute;
    
    bottom: -200px;
    left: -40px;
    background-attachment: fixed;
  }
  .scoreboard_section > .scoreboard__title::before {
    content: "";
    /* width: 10%; */
    height: 40%;
    padding-left: 10px;
    background: inherit;
    position: absolute;
    top: 0px;
    /* left: 200px; */
    right: 50px;
    background-attachment: fixed;
    -webkit-filter: blur(12px);
    filter: blur(12px);
  }
  .DashboardList_list {
    width: 100%;
  }

  .watchlist_main {
      width: 100%;
  }

  ul {
    /* display: flex; */
    list-style-type: disc;
    margin-block-start: 0em;
    margin-block-end: 0em;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
    padding-inline-start: 0px;
}

.scoreboard_section {
    text-align: center;
    /* display: flex; */
    height: 70vh;
    padding-bottom: 0;
    width: 100%;
    background: linear-gradient(
        to bottom,
        rgba(62, 8, 71, 0.979),
        rgba(216, 13, 131, 0.6)
      ), url('https://i.imgur.com/gHdH0or.png');
    background-size: contain;
    background-repeat: no-repeat;
    font-size: 15px;
    /* text-align: left; */
    padding-left: 90px;
}
.score__div {
    top: 50px;
    position: relative;
}


.scoreboard_list_section {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-content: space-between;
    margin-top: 50px;
    width: 70%;
    margin-left: 15px;
}

h2.scoreboard__title {
    padding: 0px 0px;
    margin-block-start: 0em;
}


.DashboardList_list {
    overflow: auto;
    width: 100%;
    height: 500px;
    font-size: 15px;
    padding-bottom: 20px;
    text-align: left;
}

.list__title {
    text-align: center;
    
}

}

@media only screen and (min-device-width: 319px) and (max-device-width: 568px) 
and (-webkit-device-pixel-ratio: 2){
    .scoreboard_section > .scoreboard__title {
        top: 100px;
        width: 100%;
    }
    .scoreboard_section > .scoreboard__title::before {
        width: 100%;
    }

    .scoreboard__title {
        margin-top: 0px;
        margin-left: 100px;
    }
}

@media (min-width: 768px) and (max-width: 1270px) {
    .DashboardListNav {
        display: flex;
        flex-direction: column;
        height: 100vh;
    }
    .DashboardList_list {
        width: 100%;
        height: 100vh;
        
    }
    .watchlist_main {
        font-size: 20px;
        margin-bottom: 10px;
        
        width: 95%;
    }

    .scoreboard_section > .scoreboard__title {
        width: 91%;
        padding: 20px 10px;
        background: inherit;
        background-attachment: fixed;
      }
      .scoreboard_section > .scoreboard__title::before {
        content: "";
        height: 100%;
        background: inherit;
        background-attachment: fixed;
        -webkit-filter: blur(12px);
        filter: blur(12px);
      }
      .scoreboard_section {
          padding-top: 20px;
          background-size: contain;
          background-position: 60%;
          align-content: center;
          display: flex;
          flex-direction: column;
          align-items: center;
          width: 100%;
          height: 100vh;
          padding-right: 50px;
      }

      .scoreboard_list_section {
          width: 67%;
          font-size: 20px;
          margin-right: 120px;
      }

      .scoreboard__title {
          margin-top: 25px;
          width: 100%;
          margin-bottom: 20px;
      }
      li.scoreboard__list {
        margin: 21px;
    }
}

@media (min-width: 1271px) and (max-width: 1767px) {
    .scoreboard_title {
        width: 100%;
        position: relative;
    }
    .scoreboard_section > .scoreboard__title {
        width: 100%;
        background: inherit;
        background-attachment: fixed;
    }
    .scoreboard_list_section {
        width: 110%;
        display: flex;
        flex-direction: column;
        margin-left: -100px;
    }

    .scoreboard_section {
        padding-right: 10px;
    }
}

@media only screen and (min-device-width: 319px) and (max-device-width: 568px) 
and (-webkit-device-pixel-ratio: 2){
    .DashboardListNav {
        height: 100vh;
        display: flex;
        flex-direction: column;
    }

    .DashboardList_list {
        display: flex;
        flex-direction: column;
        font-size: 15px;
        width: 100%;
        height: 50vh;
    }
    .list__title {
        font-size: 16px;
        width: 100%;
    }
    .watchlist_main {
        width: 100%;
        margin-right: 45px;
    }


    .scoreboard_section {
        background-size: contain;
        background-position-x: center;
        background-position-y: bottom;
        width: 100%;
        text-align: center;
        left: 100px;
        height: 50vh;
        padding-right: 5px;
    }

    .score__div {
        position: relative;
        top: 100px;
        left: -50px;
        font-size: 15px;
    }
    .scoreboard__title {
        font-size: 15px;
        left: -100px;
        margin-top: -50px;
        position: relative;
    }
    .scoreboard_section > .scoreboard__title {
        width: 100%;
        padding: 20px 10px;
        background: inherit;
        background-attachment: fixed;
      }
      .scoreboard_section > .scoreboard__title::before {
        content: "";
        width: 100%;
        height: 100%;
        background: inherit;
        background-attachment: fixed;
        -webkit-filter: blur(12px);
        filter: blur(12px);
      }
}