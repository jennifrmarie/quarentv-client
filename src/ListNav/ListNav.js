import React, { Component } from 'react'
import moment from 'moment'
import AppContext from '../AppContext'
import './ListNav.css'
import config from '../config'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export default class ListNav extends Component {
    static contextType = AppContext

    handleUnwatchButton = (entryId) => {
        this.updateScore(-100)
        this.context.unwatchEntry(entryId)
        this.context.editItem(entryId, {
            watched: 0,
        })
    }

    handleWatchButton = (entryId) => {
        this.updateScore(100)
        this.context.editItem(entryId, {
            watched: 1,
        })
        this.context.removeEntry(entryId)
    }

    updateScore(points) {

        const newScore = this.context.score + points
        
        return fetch(`${config.API_ENDPOINT}/users/score`, {
            method: 'post',
            headers: {
                'content-type': 'application/json',
                "Authorization": `Bearer ${localStorage.authToken}`,
            },
            
            body: JSON.stringify({
                score: newScore,
            })
            
        })
        .then(entry => {
            this.context.addScore(newScore)
          
          })
        .catch(error => {
          alert(JSON.stringify(error)) 
            
        })
    }
    render() {
        
        return (
            <div className="DashboardListNav">
                <ul className='DashboardList_list'>
                    <h2 className="list__title">Get to work!</h2>
                    {this.context.entries.map(entry =>
                        <li key={entry.id} className="watchlist_main">
                            {entry.title}{' '}
                            {entry.strmservice}{' '}
                            {moment(entry.date).format('M-D-YY')}
                            <div className="delete_button" onClick={e => this.context.removeEntry(entry.id)}><FontAwesomeIcon icon="trash-alt"></FontAwesomeIcon></div>
                            <div className="checkbox_input">
                                <input type="checkbox" onClick={e => this.handleWatchButton(entry.id)} name='checkbox' id='checkbox' />
                                <label htmlFor="watched">watched</label>
                                
                            </div>
                            
                            
                        </li>)}
                </ul>
                
                <section className="scoreboard_section">
                    <h2 className="scoreboard__title">Scoreboard</h2>
                    <ul className="scoreboard_list_section">
                        <div>Score: {this.context.score}</div>{'\n'}
                        {this.context.userBadges.map(badge => (
                        <li className="scoreboard__list">
                            {this.context.badgesMessage}{'\n'}
                            {badge.img}
                        </li>
                         ))}{'\n'}
                        {this.context.nextMessage}
                    </ul>  
                </section>
                  {/* <section className="watched_list"> */}
                  {/* <WatchList 
                    handleUnwatchButton = {this.handleUnwatchButton}
                    /> */}
                {/* </section> */}
            </div>
        )
    }
}
