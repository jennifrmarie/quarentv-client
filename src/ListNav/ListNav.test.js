import React from 'react';
import ReactDOM from 'react-dom';
import ListNav from './ListNav';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ListNav />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ListNav name="Messages" unread={4}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });