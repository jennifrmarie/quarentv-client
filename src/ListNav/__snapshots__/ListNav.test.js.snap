// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="DashboardListNav"
>
  <ul
    className="DashboardList_list"
  >
    <h2
      className="list__title"
    >
      Get to work!
    </h2>
  </ul>
  <section
    className="scoreboard_section"
  >
    <h2
      className="scoreboard__title"
    >
      Scoreboard
    </h2>
    <ul
      className="scoreboard_list_section"
    >
      <div>
        Score: 
        0
      </div>
      

      

    </ul>
  </section>
</div>
`;
